x/mongo/driver/topology/errors.go

```diff
@@ -95,14 +102,23 @@ func (w WaitQueueTimeoutError) Error() string {
 		)
 	}
 
+	var msg string
+	openConnectionCount := uint64(w.totalConnectionCount)
+	if pinnedConnections := w.pinnedConnections; pinnedConnections != nil {
+		openConnectionCount -= (*pinnedConnections).cursorConnections
+		msg += fmt.Sprintf("connections in use by cursors: %d, ", (*pinnedConnections).cursorConnections)
+		openConnectionCount -= (*pinnedConnections).transactionConnections
+		msg += fmt.Sprintf("connections in use by transactions: %d, ", (*pinnedConnections).transactionConnections)
+	}
 	return fmt.Sprintf(
-		"%s; maxPoolSize: %d, connections in use by cursors: %d"+
-			", connections in use by transactions: %d, connections in use by other operations: %d",
+		"%s; maxPoolSize: %d, %sconnections in use by other operations: %d, idle connections: %d, wait duration: %s",
```

Comment:
Should we also include the "connections in use by other operations" portion in the pinnedConnection branch?

```go
func (w WaitQueueTimeoutError) Error() string {
	errorMsg := "timed out while checking out a connection from connection pool"
	switch w.Wrapped {
	case nil:
	case context.Canceled:
		errorMsg = fmt.Sprintf(
			"%s: %s",
			"canceled while checking out a connection from connection pool",
			w.Wrapped.Error(),
		)
	default:
		errorMsg = fmt.Sprintf(
			"%s: %s",
			errorMsg,
			w.Wrapped.Error(),
		)
	}

	msg := fmt.Sprintf("%s; maxPoolSize: %d, ", errorMsg, w.maxPoolSize)

	if pinnedConnections := w.pinnedConnections; pinnedConnections != nil {
		openConnectionCount := uint64(w.totalConnectionCount) -
			(*pinnedConnections).cursorConnections -
			(*pinnedConnections).transactionConnections

		msg += fmt.Sprintf("connections in use by cursors: %d, ", (*pinnedConnections).cursorConnections)
		msg += fmt.Sprintf("connections in use by transactions: %d, ", (*pinnedConnections).transactionConnections)
		msg += fmt.Sprintf("connections in use by other operations: %d, ", openConnectionCount)
	}

	msg += fmt.Sprintf("idle connections: %d, ", w.availableConnectionCount)
	msg += fmt.Sprintf("wait duration: %s", w.waitDuration.String())

	return msg
}

```

Reply:
You are correct. The wording `connections in use by other operations` without its context is even more confusing.